### DISCUSSION TITLE
Trying to follow flet.app

### DISCUSSION THREAD

**Discussion by kkinder** [flet-dev/flet#707] [Q&A] [answered]
Hi, I'm trying to follow flet.app usage. When I see examples you give for _production_ environments, I see that you do something like this:

```
ft.app(port=os.environ['PORT'], target=main)
```

I find this confusing however, because when I run it locally, it starts a native window to interact with. If I run this inside a Docker environment, it just automatically knows not to? Does Flet know whether it's in dev mode or not without any parameters? How can I be sure it's in a production setting if I'm not passing anything?

Cheers!

-Ken

**Comment by FeodorFitsner** [comment, marked as answer]
That `port=os.environ['PORT']` is only needed because `PORT` is a hardcoded variable passed by a hosting environment. Fletd (server) listening port can be set with `FLET_SERVER_PORT` environment variable. Let me know if I got your question right.

**Reply by kkinder** [reply]
Yeah, that makes sense, but I guess I'm more asking whether Flet knows whether it's in a production or development environment. When I run it interactively, it shows up with an app window. If I pass `view=flet.WEB_BROWSER`, it opens a browser. Both of those seem like development environment profiles (opening a window, or a browser).

Does Flet just know when it's in a production environment and not do those things?

Thanks!
-Ken
